openwisp_monitoring/monitoring/static/monitoring/js/chart.js

```diff
@@ -16,6 +16,75 @@
         }
         return newArr;
     }
+
+    function adaptiveFilterPoints(charts, layout, unit, data) {
+        var newArr = charts[0].y, sum = 0, count = 0, multiplier = 1;
+
+        for(var i=0; i<newArr.length; i++) {
+            sum += newArr[i];
+        }
+        for(i=0; i<newArr.length; i++) {
+            if(newArr[i] !== 0) {
+                count++;
+            }
+        }
+
+        var average = sum / count;
+        if (average < 0.01) {
+            for(i=0; i<newArr.length; i++) {
+                multiplier = 1000000;
+                charts[0].y[i] = Math.round((charts[0].y[i] * multiplier) * 100) / 100;
+                charts[1].y[i] = Math.round((charts[1].y[i] * multiplier) * 100) / 100;
+                charts[2].y[i] = Math.round((charts[2].y[i] * multiplier) * 100) / 100;
+            }
+            layout.yaxis.title = 'KB';
+            unit = 'KB';
+        }
+        else if (average < 1) {
+            for(i=0; i<newArr.length; i++) {
+                multiplier = 1000;
+                charts[0].y[i] = Math.round((charts[0].y[i] * multiplier) * 100) / 100;
+                charts[1].y[i] = Math.round((charts[1].y[i] * multiplier) * 100) / 100;
+                charts[2].y[i] = Math.round((charts[2].y[i] * multiplier) * 100) / 100;
```

Comment:
isn't the same operation repeated multiple times here and above? 

What about:

```javascript
adaptValue = function(value, multiplier) {
  return Math.round((value * multiplier) * 100) / 100;
}
```

Reply:
Yeah! I will improve this